### PR TITLE
TSS-1364: Upgrade barrier history to use V2

### PR DESCRIPTION
**Pull request by abarolo**

### CODE REVIEW

File: api/history/manager.py [hunk 1]
```diff
@@ -107,18 +135,15 @@ def get_full_history(cls, barrier, ignore_creation_items=False, use_cache=False)
...
-        history += cls.get_economic_impact_assessment_history(
-            barrier.pk, start_date=start_date
-        )
-        history += cls.get_resolvability_assessment_history(
-            barrier.pk, start_date=start_date
-        )
-        history += cls.get_strategic_assessment_history(
-            barrier.pk, start_date=start_date
-        )
+        # history += cls.get_economic_impact_assessment_history(
```
Comment (Santino-Trade) [inline api/history/manager.py:138]: remove old stuff

File: api/assessment/models.py [hunk 2]
```diff
@@ -114,6 +116,30 @@ def save(self, *args, **kwargs):
...
+        default_fields = (
+            "approved",
+            "archived",
+            "documents_cache",
+            "explanation",
+            "export_value",
+            "import_market_size",
+            "rating",  # Needs enrichment
+            "ready_for_approval",
+            "value_to_economy",
```
Comment (Santino-Trade) [inline api/assessment/models.py:131]: Do we need all these fields? Are they all sucked up by the frontend and not used, or do we display all of these fields on the history cards? (Same question for the other models)

File: api/assessment/models.py [hunk 3]
```diff
@@ -114,6 +116,30 @@ def save(self, *args, **kwargs):
...
+    def get_history(cls, barrier_id: str, fields: Optional[List] = None):
+        qs = cls.history.filter(barrier__id=barrier_id)
+        default_fields = (
+            "approved",
+            "archived",
+            "documents_cache",
+            "explanation",
+            "export_value",
+            "import_market_size",
+            "rating",  # Needs enrichment
```
Comment (Santino-Trade) [inline api/assessment/models.py:129]: remove enrichment reminders

File: api/assessment/models.py [hunk 4]
```diff
@@ -1,3 +1,4 @@
+from typing import List, Optional
```
Comment (Santino-Trade) [inline api/assessment/models.py:1]: Easier to see with a comment here, but theres a failing U test

Comment (abarolo) [reply api/assessment/models.py:1]: That test will be redundant https://github.com/uktrade/market-access-api/pull/676

**Pull request merged**